bootstrap/src/main/java/io/airlift/bootstrap/LifeCycleManager.java

```diff
@@ -110,14 +110,52 @@ public void start()
 
     /**
      * Stop the life cycle - all instances will have their {@link javax.annotation.PreDestroy} method(s) called
+     * and any exceptions raised will be collected and thrown in a wrapped {@link LifeCycleStopException} as
+     * suppressed exceptions. Those failures will not be logged and are the responsibility of the caller to
+     * handle appropriately.
      *
-     * @throws Exception errors
+     * @throws LifeCycleStopException If any failure occurs during the clean up process
+     */
+    public void stopWithoutFailureLogging()
+            throws LifeCycleStopException
+    {
+        List<Throwable> failures = stopInternal(false);
+        if (!failures.isEmpty()) {
+            LifeCycleStopException e = new LifeCycleStopException();
+            for (Throwable t : failures) {
+                e.addSuppressed(t);
+            }
+            throw e;
+        }
+    }
+
+    /**
+     * Stop the life cycle - all instances will have their {@link javax.annotation.PreDestroy} method(s) called
```

Comment:
`PreDestroy` probably doesn't need to be qualified, since it's already imported. Same for other usages.